src/ui.rs

```diff
@@ -88,36 +97,62 @@ pub struct UiCell<'a> {
+    ///
+    /// - theme: Theme::default()
+    /// - maybe_widgets_capacity: None
+    pub fn new() -> Self {
+        UiBuilder {
+            maybe_theme: None,
+            maybe_widgets_capacity: None
+        }
+    }
 
-impl Ui {
+    /// Sets the theme of built **Ui**.
+    ///
+    /// This method moves the current builder,
+    /// unlike [`set_theme()`](#method.set_theme),
+    /// so that it can be used in method chaining.
+    pub fn theme(mut self, value: Theme) -> Self {
+        self.maybe_theme = Some(value);
+        self
+    }
 
-    /// A new, empty **Ui**.
-    pub fn new(theme: Theme) -> Self {
-        let widget_graph = Graph::new();
-        let depth_order = graph::DepthOrder::new();
-        let updated_widgets = std::collections::HashSet::new();
-        Self::new_internal(theme, widget_graph, depth_order, updated_widgets)
+    /// Sets widgets capacity of built **Ui**.
+    ///
+    /// This method moves the current builder,
+    /// unlike [`set_widgets_capacity()`](#method.set_widgets_capacity),
+    /// so that it can be used in method chaining.
+    pub fn widgets_capacity(mut self, value: usize) -> Self {
+        self.maybe_widgets_capacity = Some(value);
+        self
     }
 
-    /// A new **Ui** with the capacity given as a number of widgets.
-    pub fn with_capacity(theme: Theme, n_widgets: usize) -> Self {
-        let widget_graph = Graph::with_node_capacity(n_widgets);
-        let depth_order = graph::DepthOrder::with_node_capacity(n_widgets);
-        let updated_widgets = std::collections::HashSet::with_capacity(n_widgets);
-        Self::new_internal(theme, widget_graph, depth_order, updated_widgets)
+    /// Build **Ui** from the given builder
+    pub fn build(self) -> Result<Ui, String> {
+        Ui::new(self)
     }
+}
+
+impl Ui {
+
+    /// A new, empty **Ui**.
+    fn new(builder: UiBuilder) -> Result<Self, String> {
+        let (mut widget_graph, depth_order, updated_widgets) =
+            builder.maybe_widgets_capacity.map_or_else(
+                || (Graph::new(),
+                   graph::DepthOrder::new(),
+                   std::collections::HashSet::new()),
+                |n| (Graph::with_node_capacity(n),
+                     graph::DepthOrder::with_node_capacity(n),
+                     std::collections::HashSet::with_capacity(n)));
 
-    /// An internal constructor to share logic between the `new` and `with_capacity` constructors.
-    fn new_internal(theme: Theme,
-                    mut widget_graph: Graph,
-                    depth_order: graph::DepthOrder,
-                    updated_widgets: std::collections::HashSet<NodeIndex>) -> Self
-    {
         let window = widget_graph.add_placeholder();
         let prev_updated_widgets = updated_widgets.clone();
-        Ui {
+        Ok(Ui {
             widget_graph: widget_graph,
-            theme: theme,
+            theme: builder.maybe_theme.unwrap_or(Theme::default()),
```

Comment:
This line always constructs the `Theme::default()`, even if `maybe_theme` is `Some`.

Instead, a more efficient alternative here would be: 
```rust
theme: builder.maybe_theme.unwrap_or_else(Theme::default),
```

Here, the function `Theme::default` is instead only called if `maybe_theme` is `None`.